libwallet/src/key_pair.rs

```diff
@@ -24,12 +26,13 @@ impl<const N: usize> Signature for Bytes<N> {}
 /// Something that can sign messages
 pub trait Signer {
     type Signature: Signature;
-    fn sign_msg<M: AsRef<[u8]>>(&self, msg: M) -> Self::Signature;
-    fn verify<M: AsRef<[u8]>>(&self, msg: M, sig: &[u8]) -> bool;
+    async fn sign_msg<M: AsRef<[u8]>>(&self, data: M) -> Result<Self::Signature, ()>;
```

Comment:
```suggestion
    async fn sign_msg(&self, data: impl AsRef<[u8]>) -> Result<Self::Signature, ()>;
```
I think when I started it the syntax wasn't stable or well known